File: Survey2-Functional/factorial.hs
-- Function: Fibonacci w/ Tail-Recursion
-- Team: Haskers
-- Author: Ryan Rosier
-- Reviewer:

fac n =
    factorial n 1

factorial x a =
	if x > 1
        then factorial (x-1) (a*x)
        else a

main = print (fac 4)

File: Survey2-Functional/factorial.java
public class factorial{

    public static void main(String[] args){
        System.out.println(fac(4));
    }

    public static int fac(int n){
        return factorial(n,1);
    }

    public static int factorial(int n, int a){
        int aa = a*n;
        if(n > 2){
            return factorial(n-1, aa);
        }
        else{
            return aa;
        }
    }
}
